frontend/src/components/MeetingRoom/LayoutButton/LayoutButton.tsx

```diff
@@ -6,23 +6,30 @@ import useSessionContext from '../../../hooks/useSessionContext';
 import ToolbarButton from '../ToolbarButton';
 import useIsSmallViewport from '../../../hooks/useIsSmallViewport';
 
-type LayoutButtonProps = {
+export type LayoutButtonProps = {
   isScreenSharePresent: boolean;
+  isPinnedPresent: boolean;
 };
 
 /**
  * LayoutButton Component
  *
  * Displays a button to toggle the meeting room layout for the user between `grid` and `active-speaker`.
- * @param {boolean} isScreenSharePresent - Indicates whether there is a screenshare currently in the session.
+ * @param {LayoutButtonProps} props - the props for the component.
+ *  @property {boolean} isScreenSharePresent - Indicates whether there is a screenshare currently in the session.
+ *  @property {boolean} isPinnedPresent - Indicates whether there is a participant currently pinned.
  * @returns {ReactElement} The LayoutButton component.
  */
-const LayoutButton = ({ isScreenSharePresent }: LayoutButtonProps): ReactElement | false => {
+const LayoutButton = ({
+  isScreenSharePresent,
+  isPinnedPresent,
+}: LayoutButtonProps): ReactElement | false => {
   const { layoutMode, setLayoutMode } = useSessionContext();
   const isGrid = layoutMode === 'grid';
+  const isDisabled = isScreenSharePresent || isPinnedPresent;
 
   const handleClick = () => {
-    if (isScreenSharePresent) {
+    if (isDisabled) {
```

Comment:
Do we have test coverage for this?  That covers both screen share and pinning?

Reply:
I left a note here: https://github.com/Vonage/vonage-video-react-app/pull/115#discussion_r1984150059